url933dd5727b_AppleLiang_Testing_Project_tgz-pJ8-org_joda_time_chrono_gj_MainTestJ8/src/org/joda/time/TestDateTimeZoneCutover.java

```diff
@@ -0,0 +1,1340 @@
+        new DateTime(2007, 4, 1, 5, 0, 0, 0, MOCK_TURK);
+        new DateTime(2007, 4, 1, 6, 0, 0, 0, MOCK_TURK);
+    }
+
+    public void test_DateTime_parse_Turk() {
+        try {
+            new DateTime("2007-04-01T00:00", MOCK_TURK);
+            fail();
+        } catch (IllegalInstantException ex) {
+            assertEquals(true, ex.getMessage().indexOf("Illegal instant due to time zone offset transition") >= 0);
+        }
+    }
+
+    //-----------------------------------------------------------------------
+    //-----------------------------------------------------------------------
+    //-----------------------------------------------------------------------
+    /** America/New_York cutover from 01:59 to 03:00 on 2007-03-11 */
+    private static long CUTOVER_NEW_YORK_SPRING = 1173596400000L;  // 2007-03-11T03:00:00.000-04:00
+    private static final DateTimeZone ZONE_NEW_YORK = DateTimeZone.forID("America/New_York");
+//  DateTime x = new DateTime(2007, 1, 1, 0, 0, 0, 0, ZONE_NEW_YORK);
+//  System.out.println(ZONE_NEW_YORK.nextTransition(x.getMillis()));
+//  DateTime y = new DateTime(ZONE_NEW_YORK.nextTransition(x.getMillis()), ZONE_NEW_YORK);
+//  System.out.println(y);
+
+    //-----------------------------------------------------------------------
+    public void test_NewYorkIsCorrect_Spring() {
+        DateTime pre = new DateTime(CUTOVER_NEW_YORK_SPRING - 1L, ZONE_NEW_YORK);
+        assertEquals("2007-03-11T01:59:59.999-05:00", pre.toString());
+        DateTime at = new DateTime(CUTOVER_NEW_YORK_SPRING, ZONE_NEW_YORK);
+        assertEquals("2007-03-11T03:00:00.000-04:00", at.toString());
+        DateTime post = new DateTime(CUTOVER_NEW_YORK_SPRING + 1L, ZONE_NEW_YORK);
+        assertEquals("2007-03-11T03:00:00.001-04:00", post.toString());
+    }
+
+    public void test_getOffsetFromLocal_NewYork_Spring() {
+        doTest_getOffsetFromLocal(3, 11, 1, 0, "2007-03-11T01:00:00.000-05:00", ZONE_NEW_YORK);
+        doTest_getOffsetFromLocal(3, 11, 1,30, "2007-03-11T01:30:00.000-05:00", ZONE_NEW_YORK);
+        
+        doTest_getOffsetFromLocal(3, 11, 2, 0, "2007-03-11T03:00:00.000-04:00", ZONE_NEW_YORK);
+        doTest_getOffsetFromLocal(3, 11, 2,30, "2007-03-11T03:30:00.000-04:00", ZONE_NEW_YORK);
+        
+        doTest_getOffsetFromLocal(3, 11, 3, 0, "2007-03-11T03:00:00.000-04:00", ZONE_NEW_YORK);
+        doTest_getOffsetFromLocal(3, 11, 3,30, "2007-03-11T03:30:00.000-04:00", ZONE_NEW_YORK);
+        doTest_getOffsetFromLocal(3, 11, 4, 0, "2007-03-11T04:00:00.000-04:00", ZONE_NEW_YORK);
+        doTest_getOffsetFromLocal(3, 11, 5, 0, "2007-03-11T05:00:00.000-04:00", ZONE_NEW_YORK);
+        doTest_getOffsetFromLocal(3, 11, 6, 0, "2007-03-11T06:00:00.000-04:00", ZONE_NEW_YORK);
+        doTest_getOffsetFromLocal(3, 11, 7, 0, "2007-03-11T07:00:00.000-04:00", ZONE_NEW_YORK);
+        doTest_getOffsetFromLocal(3, 11, 8, 0, "2007-03-11T08:00:00.000-04:00", ZONE_NEW_YORK);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 498:510, 829:841.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.